### DISCUSSION TITLE
Has anyone actually managed to install Zitadel self-hosted?

### DISCUSSION THREAD

**Discussion by idc77** [zitadel/zitadel#8166] [General]
The documentation is the worst.
I think it's so people are driven towards using the "cloud" offer, which is absolutely out of the question.

From the get started page, some env vars are set and you're supposed to know what they mean.
Why the hell does it need postgres admin access, why does it need to create a user?
Why can't I just pre-create the user and not supply the admin credentials?

I'm so annoyed.
Then you move on and are freaking slammed with configuration yaml files.

**Comment by fforootd** [comment]
Hi

I am sorry you feel that way, but tools that provide many options can feel like this from time to time.

If you share with me how you want to deploy zitadel I might be able to give you some guidance.

Btw. we also have a greate community chat on discord under https://zitadel.com/chat.

**Comment by NewtTheWolf** [comment]
I managed to install it for Self Hosted, you need some time to get into it (i needed like 3 hours for my first installation)

if you want i can share you my Production setup with an external DB Server and a custom configuration ^^

**Reply by AmirhoseinBrz** [reply]
Hello, I want to use your experience and ask a question
I implement Zitadel using docker compose and this is what I want to do
Without using the Zitadel dashboard
I should create a machine user and consider a creditional for it and manage the api using that
What is your suggestion?
I want all the steps to be without using UI and only with a series of flags in the settings 
(I was able to create a user machine for it, but unfortunately it is only created 
You must login to generate creditional from that part 
I don't want this)

**Reply by teknikscsl** [reply]
> I managed to install it for Self Hosted, you need some time to get into it (i needed like 3 hours for my first installation)
> 
> if you want i can share you my Production setup with an external DB Server and a custom configuration ^^

Please share the setup. I started going down the rabbit hole but cannot use my sub-domain.domain.com.

**Comment by patheard** [comment]
It it helps, here's some Terraform that will run it self-hosted on AWS in an ECS Fargate cluster:
https://github.com/patheard/aws-ecs-zitadel

The [docker-compose setup](https://zitadel.com/docs/self-hosting/deploy/compose) was very helpful for seeing how everything went together.

**Comment by HungryHowies** [comment]
I built my from source  works great but  does take a few minutes to understand the concept  how it supposed to  work.   Downloading the 
 Zitadel executable  with PostgreSQL was easier  to setup then CockroachDB.
